tests/src/JIT/opt/InstructionCombining/DivToMul.cs

```diff
@@ -0,0 +1,187 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+// JIT is able to replace "x / 2" with "x * 0.5" where 2 is a power of two float
+// Make sure this optimization doesn't change the results
+
+public class Program
+{
+    private static int resultCode = 100;
+
+    public static int Main(string[] args)
+    {
+        // Some corner cases
+        var testValues = new List<double>(new []
+            {
+                0, 0.01, 1.333, 1/3.0, 0.5, 1, 2, 3, 4,
+                MathF.PI, MathF.E, Math.PI, Math.E,
+                float.MinValue, double.MinValue,
+                float.MaxValue, double.MaxValue,
+                int.MaxValue, long.MaxValue,
+                int.MinValue, long.MinValue,
+                float.NegativeInfinity, double.NegativeInfinity,
+                float.PositiveInfinity, double.PositiveInfinity,
+                float.NaN, double.NaN
+            });
+
+
+        // Also, add some random values
+        var random = new Random();
+        for (int i = 0; i < 100000; i++)
+            testValues.Add(random.NextDouble());
```

Comment:
Let's hope that the code is actually correct and the test doesn't start failing randomly...

Reply:
IMO it is a really bad thing to have tests in the CI system that use `Random` without an explicit seed. My preference is to:
- provide known values to cover the variety of cases (as you've done above)
- use Random with an explicit seed to add additional coverage

At a *very* minimum (and this is *not* my preference), a pseudo-random seed can be obtained and both provided to the constructor as well as printed with a failure message when the test doesn't pass.
cc @dotnet/jit-contrib for additional opinions on this.

Reply:
Makes sense! Added a fixed seed. Btw, it looks like there are other places in other tests where Random is used with it.

Reply:
I would suggest to use `TestLibrary.Generator` instead of explicitly creating `Random()` https://github.com/dotnet/coreclr/blob/master/tests/src/Common/CoreCLRTestLibrary/Generator.cs

We've been using `TestLibrary.Generator` in other tests in coreclr (e.g. HardwareIntrinsics)

Reply:
Replaced with `TestLibrary.Generator.GetDouble()`

Reply:
IMO using random numbers in compiler test is pointless. Typically you have a few interesting corner cases and the rest. Where the rest is something like billions of cases. Unless you take special care to skew the random number distribution in a rather special manner you're wasting your time.

This is no different. There are almost no corner cases for this optimization.

Reply:
@mikedn In general I agree with you, I used these tests in TDD-style approach (Test First) and then got a lot of failures during implementation. I can delete them (at least the Random part)

Reply:
Deleting them seems reasonable to me.

Reply:
Deleted the Random part, or you suggest to delete them all?

Reply:
I only meant to delete the Random part - thanks.